packages/serverless/src/apollo/app/document.ts

```diff
@@ -32,6 +31,9 @@ import { Permission } from '@zkdb/permission';
 import { GraphqlHelper } from '@helper';
 import { DEFAULT_PAGINATION } from '@common';
 
+// The value will be validated against the schema type in the database later
+const JOI_DOCUMENT_CREATE = Joi.object().pattern(Joi.string(), Joi.any());
+
```

Comment:
Using the function to cover like this

```ts
export const collectionName = (required: boolean = true) => {
  const joiCollectionName = Joi.string()
    .trim()
    .min(4)
    .max(128)
    .pattern(/^[a-z]+[_a-z0-9]+/i);
  return required ? joiCollectionName.required() : joiCollectionName.optional();
};
```

Reply:
I don't think we need to do this, joi's builder methods clone by default
https://github.com/hapijs/joi/blob/239ec33/lib/base.js#L747

Reply:
We can't depend on third party, it would create a big problem when they change the API without our noticing

Reply:
fixed